drivers/imx6-enet.c

```diff
@@ -838,9 +993,16 @@ static int enet_netifInit(struct netif *netif, char *cfg)
 		return err;
 
 	if (cfg) {
-		err = ephy_init(&priv->phy, cfg, enet_setLinkState, (void*) priv->netif);
+		uint8_t board_rev = enet_readBoardRev();
+
+		err = ephy_init(&priv->phy, cfg, board_rev, enet_setLinkState, (void *)priv->netif);
 		if (err)
 			enet_printf(priv, "WARN: PHY init failed: %s (%d)", strerror(err), err);
+
+		err = enet_phySelfTest(netif);
```

Comment:
Does it make sense to run selftest if `ephy_init` fails?

Reply:
Probably not but I wanted to avoid increasing cyclomatic complexity. Added explicit early return.